public/js/hipsterList.js

```diff
@@ -0,0 +1,114 @@
+
+function HipsterList() {
+//menu items
+ this.shopInfo =
+      {
+        "shopName": "The Coffee Connection",
+        "address": "123 Lakeside Way",
+        "phone": "16503600708",
+        "prices": 
+          {
+            "Cafe Latte": 4.75,
+            "Flat White": 4.75,
+            "Cappucino": 3.85,
+            "Single Espresso": 2.05,
+            "Double Espresso": 3.75,
+            "Americano": 3.75,
+            "Cortado": 4.55,
+            "Tea": 3.65,
+            "Choc Mudcake": 6.40,
+            "Choc Mousse": 8.20,
+            "Affogato": 14.80,
+            "Tiramisu": 11.40,
+            "Blueberry Muffin": 4.05,
+            "Chocolate Chip Muffin": 4.05,
+            "Muffin Of The Day": 4.55
+          }
+      };
+    this.list = [];
+    this.price = [];
+    this.total = [];
+    this.quantity = [];
+    this.payment = [];
+
+}
+
+//functions for list
+  HipsterList.prototype.add = function(item, quantity) {
+    if (this.list.indexOf(item) == -1) {
+       this.list.push(item);
+       var price = (this.shopInfo.prices[item] * quantity).toFixed(2);
+       this.price.push(price);
+       this.quantity.push(quantity);
+     }
+    else {
+       var index = this.list.indexOf(item);
+       var amount = this.shopInfo.prices[item] * quantity;
+       this.price[index] = +(this.price[index]) + +(amount).toFixed(2);
+       this.quantity[index] = +this.quantity[index] + +quantity;
+    }
+  };
+  
+  HipsterList.prototype.getAllPrice = function() {
+    return this.price;
+  };
+
+  HipsterList.prototype.getAllItems = function() {
+    return this.list;
```

Comment:
if you renamed `this.list` `this.items` you wouldn't need to this method, you could just get items by calling `this.items`